Color Picker/ColorPickerScreen.swift

```diff
@@ -127,22 +127,149 @@ private struct BarView: View {
+                        } else {
+                            textColor = .red
+                        }
+
+                        if !appState.isPreventingUpdate {
+                            appState.updateColorsFromPanel(excludeHex: true, preventUpdate: true, color: colorPanel.color)
+                        }
+                    case .hsl:
+                        if
+                            isTextFieldFocused,
+                            !appState.isPreventingUpdate,
+                            let newColor = NSColor(cssHSLString: inputColorText.trimmingCharacters(in: .whitespaces))
+                        {
+                            colorPanel.color = newColor
+                        }
+
+                        if NSColor(cssHSLString: inputColorText.trimmingCharacters(in: .whitespaces)) != nil {
+                            textColor = .primary
+                        } else {
+                            textColor = .red
+                        }
+
+                        if !appState.isPreventingUpdate {
+                            appState.updateColorsFromPanel(excludeHSL: true, preventUpdate: true, color: colorPanel.color)
+                        }
+                    case .rgb:
+                        if
+                            isTextFieldFocused,
+                            !appState.isPreventingUpdate,
+                            let newColor = NSColor(cssRGBString: inputColorText.trimmingCharacters(in: .whitespaces))
+                        {
+                            colorPanel.color = newColor
+                        }
+
+                        if NSColor(cssRGBString: inputColorText.trimmingCharacters(in: .whitespaces)) != nil {
+                            textColor = .primary
+                        } else {
+                            textColor = .red
+                        }
+
+                        if !appState.isPreventingUpdate {
+                            appState.updateColorsFromPanel(excludeRGB: true, preventUpdate: true, color: colorPanel.color)
+                        }
+                    case .lch:
+                        if
+                            isTextFieldFocused,
+                            !appState.isPreventingUpdate,
+                            let newColor = NSColor(cssLCHString: inputColorText.trimmingCharacters(in: .whitespaces))
+                        {
+                            colorPanel.color = newColor
+                        }
+
+                        if NSColor(cssLCHString: inputColorText.trimmingCharacters(in: .whitespaces)) != nil {
+                            textColor = .primary
+                        } else {
+                            textColor = .red
+                        }
+
+                        if !appState.isPreventingUpdate {
+                            appState.updateColorsFromPanel(excludeLCH: true, preventUpdate: true, color: colorPanel.color)
+                        }
```

Comment:
There's a lot of duplicated logic in each case here. Would be good to reduce that.

For example, `appState.updateColorsFromPanel` could be updated to accept an enum instead of `excludeLCH: true` type parameters.